test/test-udp-send-unreachable.c

```diff
@@ -138,12 +161,54 @@ TEST_IMPL(udp_send_unreachable) {
                   send_cb);
   ASSERT(r == 0);
 
+#if defined(__linux__)
+  ASSERT(0 == uv_ip4_addr("127.0.0.1", TEST_PORT_3, &addr3));
+
+  r = uv_udp_init(uv_default_loop(), &client2);
+  ASSERT(r == 0);
+
+  r = uv_udp_bind(&client2, (const struct sockaddr*) &addr3, UV_UDP_RECVERR);
+  ASSERT(r == 0);
+
+  r = uv_udp_recv_start(&client2, alloc_cb, recv_cb);
+  ASSERT(r == 0);
+
+  /* client sends "PING", then "PANG" */
+  buf = uv_buf_init("PING", 4);
+
+  r = uv_udp_send(&req3,
+                  &client2,
+                  &buf,
+                  1,
+                  (const struct sockaddr*) &addr,
+                  send_cb_recverr);
+  ASSERT(r == 0);
+
+  buf = uv_buf_init("PANG", 4);
+
+  r = uv_udp_send(&req4,
+                  &client2,
+                  &buf,
+                  1,
+                  (const struct sockaddr*) &addr,
+                  send_cb_recverr);
+  ASSERT(r == 0);
+#endif
+
   uv_run(uv_default_loop(), UV_RUN_DEFAULT);
 
+#if defined(__linux__)
+  ASSERT(send_cb_called == 4);
+#else
   ASSERT(send_cb_called == 2);
+#endif
   ASSERT(recv_cb_called == alloc_cb_called);
   ASSERT(timer_cb_called == 1);
+#if defined(__linux__)
+  ASSERT(close_cb_called == 3);
+#else
   ASSERT(close_cb_called == 2);
+#endif
```

Comment:
Not a blocker but I'd prefer to get rid of the #ifdefs and use a flag. One or two #ifdefs aren't a problem but a lot are. What I mean is this:
```c
TEST_IMPL(udp_send_unreachable) {
  // ...
  int can_recverr;
  // ...
#ifdef __linux__
  can_recverr = 1;
#else
  can_recverr = 0;
#endif
  // ...
  if (can_recverr) {
    // ...
  }
  // ...
  ASSERT_EQ(send_cb_called, can_recverr ? 4 : 2);
  // ...
}
```
Feel free to update any ASSERT statements you touch to the newer ASSERT_EQ macro.

Reply:
@bnoordhuis Ok, I did that in a fixup commit, let me know if it's what you had on mind.

Reply:
Isn't this the case that it's actually no-op on the other platforms because the ICMP reporting is already enabled there and doesn't require a special flag?

At least the `send()` on FreeBSD already returns [ECONNREFUSED](https://www.freebsd.org/cgi/man.cgi?query=send&sektion=2&apropos=0&manpath=FreeBSD+12.1-RELEASE+and+Ports) as does `send()` on [OpenBSD](https://man.openbsd.org/sendmsg.2).